packages/ckeditor5-utils/src/dom/position.ts

```diff
@@ -119,11 +152,22 @@ export function getOptimalPosition( { element, target, positions, limiter, fitIn
 		// @if CK_DEBUG_POSITION // 	RectDrawer.draw( limiterRect, { outlineWidth: '5px', outlineColor: 'green' }, 'Visible limiter' );
 		// @if CK_DEBUG_POSITION // }
 
-		Object.assign( positionOptions, { limiterRect, viewportRect } );
+		const properRectLimiter = isNaN( ancestorsIntersectionWindowRect.top ) ? limiterRect : ancestorsIntersectionWindowRect;
+		const limiterViewportIntersection = ( properRectLimiter && !isNaN( properRectLimiter.top ) && viewportRect ) ?
+			properRectLimiter.getIntersection( viewportRect! ) : properRectLimiter;
+
+		Object.assign( positionOptions, {
+			limiterRect, viewportRect: target instanceof Range ? limiterRect : limiterViewportIntersection } );
+
+		// When `limiter` is not set and `getBestPosition` return null lets return the first `position`
+		// from the `positions` list. If the limiter is set - we will take the last `position` (hide).
+		const indexOfPositionToReturn = limiter ? -1 : 0;
```

Comment:
Like I said before, an assumption that the last available position is the one that hides the positioned element is weak.